test/testrunner.py

```diff
@@ -435,11 +438,15 @@ def main():
                     "--ignore=container_tests/",
                 ]
             ]
-            if specific_test_type == "canary":
+            if specific_test_type in ["canary", "deep-canary"]:
                 # Add rerun flag to canaries to avoid flakiness
                 pytest_cmds = [
                     pytest_cmd + ["--reruns=1", "--reruns-delay=10"] for pytest_cmd in pytest_cmds
                 ]
+            if specific_test_type == "deep-canary":
```

Comment:
It seems odd from a scoping perspective to have two same-level conditions referencing deep canaries. Any reason to not nest this under the statement above?

Reply:
No reason specifically. You are right. I'll refactor this a bit, and change the list that is added to `pytest_cmd` based on whether it is in `deep_canary` mode or not.